app-backend/tile/src/main/scala/routes/ToolRoutes.scala

```diff
@@ -1,75 +1,65 @@
 package com.azavea.rf.tile.routes
 
+import com.azavea.rf.common.Authentication
 import com.azavea.rf.tile.image._
+import com.azavea.rf.tile.tool.TileSources
+import com.azavea.rf.tile._
+import com.azavea.rf.tool.ast._
 import com.azavea.rf.database.Database
-
+import com.azavea.rf.database.tables.Tools
+import com.azavea.rf.datamodel._
+import com.azavea.rf.tool.ast.codec._
+import com.azavea.rf.tool.ast._
+import com.azavea.rf.tool.op._
+
+import io.circe._
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.generic.auto._
+import de.heikoseeberger.akkahttpcirce.CirceSupport._
 import geotrellis.raster._
 import geotrellis.raster.render._
 import geotrellis.raster.io._
 import geotrellis.raster.op._
 import geotrellis.raster.render.{Png, ColorRamp, ColorMap}
 import geotrellis.raster.io.geotiff._
 import geotrellis.vector.Extent
+import geotrellis.slick.Projected
 import geotrellis.spark._
-import geotrellis.proj4.CRS
+import geotrellis.spark.tiling._
+import geotrellis.proj4._
 import akka.http.scaladsl.server._
 import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.marshalling.Marshal
 import akka.http.scaladsl.model.{ContentType, HttpEntity, HttpResponse, MediaTypes}
-import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
 import com.typesafe.scalalogging.LazyLogging
-import spray.json._
+import cats.data._
+import cats.data.Validated._
 import cats.implicits._
+
+import java.security.InvalidParameterException
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent._
 import java.util.UUID
+import java.io._
 
-// TODO: I need a better way to bind ops ... Right now I just swap out Var for Op ...
 
-object ToolRoutes extends LazyLogging {
+class ToolRoutes(implicit val database: Database) extends Authentication with LazyLogging {
   val userId: String = "rf_airflow-user"
```

Comment:
An instance of this is being created here: https://github.com/azavea/raster-foundry/blob/67281a7266009938c71b188dd60553d56e8c6dda/app-backend/tile/src/main/scala/Router.scala#L38